autotx/utils/configuration.py

```diff
@@ -8,34 +8,78 @@
 from autotx.utils.ethereum.constants import DEVNET_RPC_URL
 from autotx.utils.ethereum.eth_address import ETHAddress
+from autotx.utils.ethereum.networks import NetworkInfo
 from autotx.utils.is_dev_env import is_dev_env
+from autotx.wallets.smart_wallet import SmartWallet
 
 smart_account_addr = get_env_vars()
 
-def get_configuration() -> tuple[ETHAddress | None, LocalAccount, EthereumClient]:
-    rpc_url = DEVNET_RPC_URL if is_dev_env() else os.getenv("CHAIN_RPC_URL")
-    
-    if not rpc_url:
-        sys.exit("CHAIN_RPC_URL is not set")
+class AppConfig:
+    web3: Web3
+    client: EthereumClient
+    agent: LocalAccount
+    manager: SafeManager
+    network_info: NetworkInfo
 
-    web3 = Web3(Web3.HTTPProvider(rpc_url))
+    def __init__(
+        self,
+        web3: Web3,
+        client: EthereumClient,
+        agent: LocalAccount,
+        manager: SafeManager,
+        network_info: NetworkInfo,
+    ):
+        self.web3 = web3
+        self.client = client
+        self.agent = agent
+        self.manager = manager
+        self.network_info = network_info
 
-    for i in range(16):
-        if web3.is_connected():
-            break
-        if i == 15:
-            if is_dev_env():
-                sys.exit("Can not connect with local node. Did you run `poetry run start-devnet`?")
-            else:
-                sys.exit("Can not connect with remote node. Check your CHAIN_RPC_URL")
-        sleep(0.5)
+    @staticmethod
+    def load(
+        smart_account_addr: str | None = None,
+        subsidized_chain_id: int | None = None, 
+    ) -> "AppConfig":
+        rpc_url: str
 
-    client = EthereumClient(URI(rpc_url))
-    agent = get_or_create_agent_account()
+        if subsidized_chain_id:
+            network_info = NetworkInfo.from_chain_id(subsidized_chain_id)
+            subsidized_rpc_url = network_info.get_subsidized_rpc_url()
+            
+            if not subsidized_rpc_url:
+                raise ValueError(f"Chain ID {subsidized_chain_id} is not supported")
+            
+            rpc_url = subsidized_rpc_url
+        else:
+            provided_rpc_url = DEVNET_RPC_URL if is_dev_env() else os.getenv("CHAIN_RPC_URL")
 
-    smart_account = ETHAddress(smart_account_addr) if smart_account_addr else None
+            if not provided_rpc_url:
+                sys.exit("CHAIN_RPC_URL is not set")
+            
+            rpc_url = provided_rpc_url
+            
+        client = EthereumClient(URI(rpc_url))
 
-    return (smart_account, agent, client)
+        for i in range(16):
+            if client.w3.is_connected():
+                break
+            if i == 15:
+                if is_dev_env():
+                    sys.exit("Can not connect with local node. Did you run `poetry run start-devnet`?")
+                else:
+                    sys.exit("Can not connect with remote node. Check your CHAIN_RPC_URL")
+            sleep(0.5)
+
+        agent = get_or_create_agent_account()
```

Comment:
In order to support user-specific agent accounts, the API caller could specify a "seed" to generate the keypair from. In the case of the discord-bot, `seed` could equal `discord_bot_secret+user_discord_id`. This also introduces the concept of a caller secret, which I think is a good thing to have for improved security.

Reply:
We need an API key per app (so discord bot will have its own) anyway, so we can use that to allows apps to create addresses based on a seed (which in this case will be the user discord id).
Here's the newly created issue: https://github.com/polywrap/AutoTx/issues/253
I think it makes sense to tackle this after the move to a DB, since we need some way of persisting both the API keys as well as the generated addresses.